geode-apis-compatible-with-redis/src/main/java/org/apache/geode/redis/internal/executor/sortedset/ZAddExecutor.java

```diff
@@ -89,6 +101,9 @@ private int findAndValidateZAddOptions(Command command, Iterator<byte[]> command
         optionsFoundCount++;
       } else if (isInfinity(subCommand)) {
         scoreFound = true;
+      } else if (isNaN(subCommand)) {
+        executorState.exceptionMessage = ERROR_NOT_A_VALID_FLOAT;
+        return 0;
```

Comment:
With the call to `Coder.bytesToDouble()` on line 60, this check shouldn't be needed, as the modified `bytesToDouble()` method will now throw an exception with the correct message when we get to that line. We would do a couple of unnecessary allocations if a user has passed NaN as a score with this removed, but we would avoid doing a byte array comparison on every argument in the more likely case where they haven't passed NaN, so it's a net performance improvement, I think.

Reply:
> With the call to `Coder.bytesToDouble()` on line 60, this check shouldn't be needed, as the modified `bytesToDouble()` method will now throw an exception with the correct message when we get to that line. We would do a couple of unnecessary allocations if a user has passed NaN as a score with this removed, but we would avoid doing a byte array comparison on every argument in the more likely case where they haven't passed NaN, so it's a net performance improvement, I think.

This comment is still relevant, I think.